src/mbgl/annotation/annotation_manager.cpp

```diff
@@ -72,6 +76,7 @@ void AnnotationManager::add(const AnnotationID& id, const StyleSourcedAnnotation
 }
 
 Update AnnotationManager::update(const AnnotationID& id, const SymbolAnnotation& annotation, const uint8_t maxZoom) {
+    assert(symbolAnnotations.count(id));
```

Comment:
`AnnotationManager::update` used to be expected to have the ability to add new annotations, but now it doesn't. Let's make this clear by changing it to:

```C++
 Update AnnotationManager::update(const AnnotationID& id, const SymbolAnnotation& annotation, const uint8_t maxZoom) {
     auto it = symbolAnnotations.find(id);
     if (it == symbolAnnotations.end()) {
         assert(false); // Attempt to update a non-existent symbol annotation
         return Update::Nothing;
     }
     ...
```